Set highlight groups for terminal mode in lightline#highlight

nhooyr opened the pull request:
Fixes #144. 
I just added the missing terminal mode to modes in lightline#highlight() if neovim is being used.
Also has('win32') includes 64bit windows. No need to run has('win64').

itchyny commented:
How the fix for win64 have something to the issue #144? Is there some reason to fix in one commit?

itchyny commented:
Also, is there some reason that we should remove `has('win64')`? For performance?

nhooyr commented:
@itchyny i just updated it a bit, now `terminal` is only added if neovim is being used. I removed `has('win64')` because `has('win32')` covers 64bit

`win32                   Win32 version of Vim (MS-Windows 95 and later, 32 or 64 bits)`

nhooyr commented:
I'll add it in a different commit.

nhooyr commented:
@itchyny what do you think now.

itchyny commented:
Thank you for your issue report and pull request.
